### PR TITLE
[13.0][IMP] sale_stock_picking_note: Add Picking comments in sale orders to use in pickings

### PR DESCRIPTION
**Pull request by victoralmau**
Add Picking comments in sale orders to use in pickings.
Migration scripts.
Add comments field to Delivery slip report.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT30766

### CODE REVIEW

File: sale_stock_picking_note/migrations/13.0.1.1.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    # Changes old values to picking_comments
```
Comment (chienandalu) [inline sale_stock_picking_note/migrations/13.0.1.1.0/post-migration.py]: This scripts assume that the current users are missusing the module, which would only be possible if the delivery slip printed the picking note field, which isn't the case.

If a user has such modification, this script should go to their custom module.

File: sale_stock_picking_note/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    comments = fields.Text(
```
Comment (chienandalu) [inline sale_stock_picking_note/models/stock_picking.py]: Better `sale_comment`?

File: sale_stock_picking_note/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+        string="Picking Comments", related="sale_id.picking_comments"
```
Comment (chienandalu) [inline sale_stock_picking_note/models/stock_picking.py]: Maybe that the warehouse user needed to add some public comments as well?

**Review by chienandalu — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Thanks @chienandalu, we will also wait for @pedrobaeza comments to avoid making changes and undoing them later.

File: sale_stock_picking_note/models/stock_picking.py (hunk 2 above)
Comment (pedrobaeza) [reply sale_stock_picking_note/models/stock_picking.py]: In `sale.order`, the field for this is called `note`. In invoices, it's called `narration`, so it seems there's no consensus about this. I think `customer_note` can be a good balance. What do you think?

File: sale_stock_picking_note/models/stock_picking.py (hunk 3 above)
Comment (pedrobaeza) [reply sale_stock_picking_note/models/stock_picking.py]: Yes, this shouldn't be a related field, but something that is propagated from sales order to picking.

File: sale_stock_picking_note/migrations/13.0.1.1.0/post-migration.py (hunk 1 above)
Comment (pedrobaeza) [reply sale_stock_picking_note/migrations/13.0.1.1.0/post-migration.py]: Indeed. Don't touch this here.

File: sale_stock_picking_note/models/stock_picking.py (hunk 2 above)
Comment (chienandalu) [reply sale_stock_picking_note/models/stock_picking.py]: Ok, I like it :)

File: sale_stock_picking_note/README.rst [hunk 4]
```diff
@@ -25,8 +25,10 @@ Sale Stock Picking Note
 
 |badge1| |badge2| |badge3| |badge4| |badge5| 
 
-This module extend sale order to include a note that will be set as the picking
+This module extend sale order to include a picking note that will be set as the picking
 note.
+This module extend sale order to include a picking comments that will be set as the picking
+comment.
```
Comment (pedrobaeza) [inline sale_stock_picking_note/README.rst]:
```suggestion
This module extends sales order to include both a external (customer) and internal picking note 
that will be transferred to the picking.
```

**Comment by victoralmau** [comment]
Thanks, changes done.

File: sale_stock_picking_note/README.rst [hunk 5]
```diff
@@ -38,8 +40,10 @@ Usage
 
 #. Place a new sale order with stockable products.
 #. Go to *Other information* tab and set the **Picking Note** you want to.
+#. Go to *Other information* tab and set the **Picking Comments** you want to.
```
Comment (pedrobaeza) [inline sale_stock_picking_note/README.rst]: Change labels for being more understandable:

```suggestion
#. Go to *Other information* tab.
#. Set the **Picking Internal Note** and **Picking Customer Comments** you want to.
```

File: sale_stock_picking_note/README.rst [hunk 6]
```diff
@@ -38,8 +40,10 @@ Usage
...
 #. Confirm the Sale Order and go to the created picking.
 #. The note is available in the "Note" tab.
+#. The comnments is available in the "Comments" tab.
```
Comment (pedrobaeza) [inline sale_stock_picking_note/README.rst]: Put both together in the same tab

```suggestion
#. On the "Comments" tab, you will see both the internal note and customer comments.
```

File: sale_stock_picking_note/__manifest__.py [hunk 7]
```diff
@@ -1,14 +1,18 @@
-# Copyright 2018 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# Copyright 2018 Tecnativa - Carlos Dauden
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 {
     "name": "Sale Stock Picking Note",
     "summary": "Add picking note in sale and purchase order",
-    "version": "13.0.1.0.0",
+    "version": "13.0.1.1.0",
```
Comment (pedrobaeza) [inline sale_stock_picking_note/__manifest__.py]: Don't change version and add your copyright

File: sale_stock_picking_note/models/sale_stock.py [hunk 8]
```diff
@@ -7,7 +8,8 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    picking_note = fields.Text(string="Picking Note",)
+    picking_note = fields.Text(string="Picking Note")
```
Comment (pedrobaeza) [inline sale_stock_picking_note/models/sale_stock.py]:
```suggestion
    picking_note = fields.Text(string="Picking Internal Note")
```

File: sale_stock_picking_note/models/sale_stock.py [hunk 9]
```diff
@@ -7,7 +8,8 @@
...
+    picking_comments = fields.Text(string="Picking Comments")
```
Comment (pedrobaeza) [inline sale_stock_picking_note/models/sale_stock.py]:
```suggestion
    picking_customer_note = fields.Text(string="Picking Customer Comments")
```

File: sale_stock_picking_note/models/stock_picking.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    customer_note = fields.Text(string="Picking Comments")
```
Comment (pedrobaeza) [inline sale_stock_picking_note/models/stock_picking.py]:
```suggestion
    customer_note = fields.Text(string="Customer Comments")
```

File: sale_stock_picking_note/views/report_deliveryslip.xml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<odoo>
+    <template
+        id="report_delivery_document_comments"
+        inherit_id="stock.report_delivery_document"
+    >
+        <xpath expr="//table[@name='stock_move_line_table']" position="after">
+            <p t-if="o.comments">
```
Comment (pedrobaeza) [inline sale_stock_picking_note/views/report_deliveryslip.xml]: This hasn't been changed. You can export to HTML the report in the tests as a little check.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_stock_picking_note/views/stock_picking_view.xml [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="view_picking_form" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <page name="note" position="after">
+                <page string="Customer Note" name="customer_note">
+                    <field
+                        name="customer_note"
```
Comment (pedrobaeza) [inline sale_stock_picking_note/views/stock_picking_view.xml]: A placeholder putting "Comments that will be printed in the delivery slip" or similar will help user.

**Review by pedrobaeza — COMMENTED** [review]
You haven't renamed the page string.

**Comment by victoralmau** [comment]
It's ready to merge, right?

**Review by chienandalu — APPROVED** [review]
:+1:

**Comment by victoralmau** [comment]
Ping @pedrobaeza

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**